### PR TITLE
Enable saving and loading preferences

### PR DESCRIPTION
**Pull request by metanMSFT**

### CODE REVIEW

File: js/src/launchAsync.ts [hunk 1]
```diff
@@ -149,7 +151,9 @@ export function launchAsync(token: string, subdomain: string, content: Content,
                     disableFirstRun: options.disableFirstRun,
                     readAloudOptions: options.readAloudOptions,
                     translationOptions: options.translationOptions,
-                    displayOptions: options.displayOptions
+                    displayOptions: options.displayOptions,
+                    sendPreferences: !!options.onPreferencesChanged,
```
Comment (rwallerms) [inline js/src/launchAsync.ts:155]: What does !! do? Make sure it is a bool?

File: js/src/options.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export type Options = {
     readAloudOptions?: ReadAloudOptions;     // Options to configure Read Aloud.
     translationOptions?: TranslationOptions; // Options to configure Translation.
     displayOptions?: DisplayOptions;         // Options to configure text size, font, etc.
+    preferences?: string;                           // String returned from onPreferencesChanged representing the user's preferences in the Immersive Reader.
```
Comment (rwallerms) [inline js/src/options.ts:18]: need UTs for these new options - null, empty, undefined, valid & invalid

File: js/src/launchAsync.ts [hunk 3]
```diff
@@ -192,6 +196,10 @@ export function launchAsync(token: string, subdomain: string, content: Content,
                     exit();
                     reject(error);
                 }
+            } else if (e.data.startsWith('ImmersiveReader-Preferences:')) {
```
Comment (rwallerms) [inline js/src/launchAsync.ts:199]: should define these message names as consts

File: js/src/launchAsync.ts [hunk 4]
```diff
@@ -192,6 +196,10 @@ export function launchAsync(token: string, subdomain: string, content: Content,
...
+                if (options.onPreferencesChanged && typeof options.onPreferencesChanged === 'function') {
+                    options.onPreferencesChanged(e.data.substring('ImmersiveReader-Preferences:'.length));
```
Comment (rwallerms) [inline js/src/launchAsync.ts:201]: consider try/catch when calling this or any user-supplied function

File: js/src/launchAsync.ts (hunk 1 above)
Comment (metanMSFT) [reply js/src/launchAsync.ts:155]: Yep, it returns true if onPreferencesChanged is not null, false otherwise.

File: js/tests/launchAsync.test.ts [hunk 5]
```diff
@@ -271,6 +271,47 @@ describe('launchAsync tests', () => {
...
+        window.postMessage('ImmersiveReader-LaunchResponse:{"success":true}', '*');
+
+        return launchPromise;
+    });
+    
+    it('launches with onPreferencesChanged set', async () => {
+        jest.useRealTimers();
+        expect.assertions(1);
+        const options: Options = { onPreferencesChanged: (value) => { 
+            expect(value).toBe('hello world');
```
Comment (rwallerms) [inline js/tests/launchAsync.test.ts:303]: can we add something to verify this actually gets called? Set a bool and check it after the await?

Comment (metanMSFT) [reply js/tests/launchAsync.test.ts:303]: the expect.assertions(1) is what verifies it

**Pull request merged**